sgkit/distance/api.py

```diff
@@ -0,0 +1,63 @@
+import numpy as np
+
+from sgkit.distance import metrics
+from sgkit.distance.core import pairwise
+from sgkit.typing import ArrayLike
+
+
+def pdist(
```

Comment:
Do we need this function?  Seems like this could be collapsed easily into `pairwise`.

Thinking about the name some more, appealing to `pdist` makes sense but it won't return squareform matrices, it will definitely have different dtype semantics, will possible handle missing values differently, will have different distance metrics, and I would guess that a lot of users won't be familiar with that in scipy anyhow.  

I'm leery of libraries that say they implement an API like that but with enough differences that it does more harm than good.  I think it would be safer to start with something generic like `pairwise` and rename to `pdist` later if we find a good reason to want greater parity with the scipy version.

Reply:
That's a very valid point, would you consider the following renaming:
`pdist` -> `pairwise` a thin wrapper around `_pairwise`
`pairwise` -> `_pairwise`

or just scrap the latter completely?

Reply:
I have removed `pdist` and merged it into `pairwise`.